### ISSUE TITLE
Tools lose durability when breaking tall grass

### ISSUE THREAD

**Issue by gigoose** [SlimeKnights/TinkersConstruct#2771]
Tested with:
Mantle 1.1.5
Tinkers' Construct 2.6.3
Forge 12.18.3.2185
no other mods.

Tools and swords lose durability when breaking instantly-breakable blocks that should not affect the durability of the tool (grass, flowers, crops, etc). Vanilla tools do not lose durability when breaking these type of blocks.

Below is the durability bar of a tool before and after 
![2017-02-12 23_33_24-minecraft 1 10 2](https://cloud.githubusercontent.com/assets/20591237/22874630/f9c4866a-f17b-11e6-8594-7d844d6625a7.png)
![2017-02-12 23_33_28-minecraft 1 10 2](https://cloud.githubusercontent.com/assets/20591237/22874629/f9c17e0c-f17b-11e6-9a26-81e1375b3184.png)

**Comment by LovelyFlare** [comment]
Not really a bug, just the "no durability loss thing" that was recently implemented in Vanilla.

**Comment by gigoose** [comment]
Just wanted to make sure that this was working as intended. In the original Tinkers Construct tools did not lose durability on these types of blocks. They only started to behave this way in Tinkers Construct 2.

Here is an example of tools not losing durability when breaking similar blocks in Tinkers Construct 1.8.8, Mantle 0.3.2b, Forge 01.13.4.1558:
![](https://i.imgur.com/8zHtvUL.png)
![](https://i.imgur.com/0M3vUd6.png)

**Comment by bonii-xx** [comment]
Sounds like a feature that got lost. On the other hand we have scythe to break large areas of grass and stuff. Hm.

**Comment by PapiDimmi** [comment]
This is a bug. Vanilla weapons and tools, along with weapons and tools from other mods, do not take damage when breaking flowers and tall grass, but Tinkers’ Construct tools and weapons do.

They should not take damage, as you can break these kinds of blocks instantly with your hand.

**Comment by bookerthegeek** [comment]
Not a bug but a feature to have you use the correct tool for the job.

I mean really, who cuts grass with a 🔨?

**Comment by PapiDimmi** [comment]
If I see a flower I want or just want to clear out some tall grass, I can easily do that no matter what I’m holding in my hand, but with Tinkers’ Construct, I’m not able to. There’s no reason to lose durability for breaking tall grass and flowers when you can _instantly_ break them no matter what you’re holding in your hand. Vanilla tools and tools added by other mods do not take damage when breaking tall grass and other types of plant, and Tinkers’ Construct tools did not take damage either in older versions, so this is undoubtedly a bug.

**Comment by PapiDimmi** [comment]
I just broke a torch with a hammer, and my hammer went down in durability by two points.

This is obviously not intended. @SlimeKnights, can you please fix this for 1.10.2?

**Comment by PapiDimmi** [comment]
@KnightMiner This is a bug report, not an enhancement request. This is clearly not intended.

**Comment by KnightMiner** [comment]
It requires being specifically programmed in and was added to vanilla as a feature. It is a feature. The distinction does not matter too much any way, it still is low priority either way.

**Comment by PapiDimmi** [comment]
It’s super annoying to have to switch tool whenever I want to break a torch, a flower, or tallgrass to avoid losing durability.

**Comment by bonii-xx** [comment]
It's also super annoying when people keep spamming the issue tracker about an issue that's already on our list and has been verified. I wonder who can be more super annoying, maybe we shouldn't fix it...?

**Comment by PapiDimmi** [comment]
What do you mean by “spamming the issue tracker”? I’m commenting on the same issue, not creating new ones.

**Comment by bonii-xx** [comment]
Comments.. Issues.. it's all the same, they all cause notifications. We have better things to do than arguing about the semantics of what is a feature/enhancement and what is a bug. From our perspective it's extra behaviour to add, hence it's not a bug, since that would imply that there's some code present that's not working as intended, so the label is correct. However having to deal with meaningless discussions like this just makes us less inclined to actually want to work on the issue.

Of course it's on our list since the issue got a proper label and is not closed yet, otherwise this wouldn't be open anymore. I also don't remember saying anything about this being a feature.

**Comment by PapiDimmi** [comment]
>Comments.. Issues.. it's all the same, they all cause notifications.

Unsubscribe from the issue and stop complaining.

**Comment by garath77** [comment]
You're a real piece of work to suggest that the developers unsubscribe from their own issue tracker. Maybe they should block you from making comments instead...

**Comment by PapiDimmi** [comment]
The only thing I did was argue with someone that this behaviour of Tinkers’ Construct is not intended. If they want to block me for that, they can go ahead and do that.

Now, this may or may not be a bug. There might be a fault in the mod’s code that makes tools lose durability when they shouldn’t, or it might just be an oversight of the mod’s developers.

**Comment by LovelyFlare** [comment]
@PapiDimmi Until you can write code yourself and push something to the issue tracker or fork it yourself, I suggest you keep silent and be patient.

**Comment by PapiDimmi** [comment]
I don’t understand how me writing code would have to do with anything. I can write code, but not in Java, and I certainly can’t create mods for Minecraft. Do I have to learn a programming language and learn how to develop Minecraft mods before I’m allowed to report a bug?

You’re telling me that unless I know how to code Minecraft mods, I’m not allowed to give feedback to Minecraft mods. That makes no sense. I did not force anyone to fix this bug—all I did was argue with someone that it’s a bug, not the intended behaviour. I don’t see what’s wrong with that. There is absolutely no need to act aggressively towards me because of that.

**Comment by bookerthegeek** [comment]
@PapiDimmi

A bug, in regards to mod developers, is something that is incorrectly coded and needs to be fixed.

A feature is something that somebody woild like to be added. As you are requesting that a function of the mod be changed to behave differently, that is why this is a feature request and not a bug.

Also, it is common to be polite to mod authors especially on there own trackers as it is their mod.

The comment about code was a not so subtle reftencw to the fact that this is not your mod, and if you would like this to get worked on in any sort of resonable timeframe you can either fix it yourself by submitting a pull request, or shutting the fuck up and stop bothering people after they have already agreed that it is something they will look into doing.

Continuing to being argumentative towards the people in charge will result in your privlage to post here being removed.

@ Everyone else,
Thanks for the wonderful mod and dont think we are all asshats please. ;)

**Comment by BrianEdmonds** [comment]
And, to be fair, some mod authors are asshats as well. (Present company 
excepted, so far as I've seen.) And apart from going around looking for 
fights, your options when you disagree with them are still to code it 
yourself, be patient, or shut up and go elsewhere. :)

**Comment by PapiDimmi** [comment]
Like I explained earlier, this could either be a bug or an oversight.

If it’s a bug, there’s an error in the code that allows tools to lose durability when they shouldn’t.
If it’s an oversight, the developers simply forgot to make tools not lose durability when breaking blocks that break instantly.

It is obvious that this is not the intended behaviour. Why should you lose durability when, say, breaking a _torch_, which you can break instantly no matter what you’re holding?

I did not attack anyone, and I did not act aggressively. All I did was argue that this is not an “enhancement” request, a flair that @KnightMiner added to this issue.

**Comment by bookerthegeek** [comment]
> It is obvious that this is not the intended behaviour. Why should you lose durability when, say, breaking a torch, which you can break instantly no matter what you’re holding?
> - PapiDimmi

It is not obvious. As stated earlier...
> ...it's extra behaviour to add, hence it's not a bug, since that would imply that there's some code present that's not working as intended...
> - bonii-xx

So if you could, please, do either of the following:
1) Point me to the part of code that is not working, as in the part of the code that currently says tools should not lose durability when breaking a torch.
2) Shut up before you piss people off with your child like behaviour. 

Also, this is not Facebook. Nobody cares about the stupid Emojis, but please feel free to continue to use them. As I feel you are not likely to understand this issue of what you are doing or how you are behaving, feel free to consider yourself blocked by myself as I am done trying to guide you in the correct direction.

**Comment by KnightMiner** [comment]
As there has been no new information and I keep getting between 2 and 4 comments on this issue everytime I check GitHub, I am locking this conversation. This will be dealt with when we have time and when we decide how it will work with scythes, more comments or convincing us this is a bug or a feature will not resolve it sooner.

**Comment by KnightMiner** [comment]
This is fixed on 1.16 in the rewrite, and I don't care to fix it on 1.12 as it will likely require significant changes to our block breaking logic.
